src/Common/Versioning/Conventions/ApiVersionConventionBuilder.cs

```diff
@@ -60,39 +60,36 @@ public virtual ControllerApiVersionConventionBuilder<TController> Controller<TCo
                 return newBuilder;
             }
 
-            if ( builder is ControllerApiVersionConventionBuilder<TController> typedBuilder )
+            if ( builder is IControllerConventionBuilder<TController> typedBuilder )
             {
                 return typedBuilder;
             }
 
+            // this should only ever happen if a subclass overrides Controller(Type) and adds a
+            // IControllerConventionBuilder that is not covariant with IControllerConventionBuilder<TController>
             throw new InvalidOperationException( SR.ConventionStyleMismatch.FormatDefault( key.Name ) );
         }
 
         /// <summary>
         /// Gets or creates the convention builder for the specified controller.
         /// </summary>
         /// <param name="controllerType">The <see cref="Type">type</see> of controller to build conventions for.</param>
-        /// <returns>A new or existing <see cref="ControllerApiVersionConventionBuilder"/>.</returns>
-        public virtual ControllerApiVersionConventionBuilder Controller( Type controllerType )
+        /// <returns>A new or existing <see cref="IControllerConventionBuilder"/>.</returns>
+        public virtual IControllerConventionBuilder Controller( Type controllerType )
         {
             Arg.NotNull( controllerType, nameof( controllerType ) );
-            Contract.Ensures( Contract.Result<ControllerApiVersionConventionBuilder>() != null );
+            Contract.Ensures( Contract.Result<IControllerConventionBuilder>() != null );
 
             var key = GetKey( controllerType );
 
             if ( !ControllerConventionBuilders.TryGetValue( key, out var builder ) )
             {
-                var newBuilder = new ControllerApiVersionConventionBuilder( controllerType );
+                var newBuilder = NewGenericControllerConventionBuilder( controllerType );
```

Comment:
`New` because it actually creates a _new_ instance. I've started to use shorted names in practice. `New` is shorter than `Create`, but _should_ have the same semantic meaning.